common/isolationgroup/defaultisolationgroupstate/state.go

```diff
@@ -81,8 +86,9 @@ func (z *defaultIsolationGroupStateHandler) AvailableIsolationGroupsByDomainID(c
 	if err != nil {
 		return nil, fmt.Errorf("unable to get isolation group state: %w", err)
 	}
-	isolationGroups := common.IntersectionStringSlice(z.config.AllIsolationGroups, availableIsolationGroups)
-	return availableIG(isolationGroups, state.Global, state.Domain), nil
```

Comment:
I moved this into inside the `availableIG` function for clarity about the decision steps that are occurring